src/Microsoft.Azure.SignalR/Startup/NegotiateMatcherPolicy.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#if NETCOREAPP3_0
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Connections;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.AspNetCore.Routing.Matching;
+using Microsoft.AspNetCore.SignalR;
+
+namespace Microsoft.Azure.SignalR.Startup
+{
+    internal class NegotiateMatcherPolicy : MatcherPolicy, IEndpointSelectorPolicy
+    {
+        // This caches the replacement endpoints for negotiate so they are not recomputed on every request
+        private readonly ConcurrentDictionary<Endpoint, Endpoint> _negotiateEndpointCache = new ConcurrentDictionary<Endpoint, Endpoint>();
+        private readonly ConcurrentDictionary<string, List<IAuthorizeData>> _hubAuthorizePolicy = new ConcurrentDictionary<string, List<IAuthorizeData>>();
```

Comment:
Remove this, we no longer need to manually apply auth policies. That *should* be done by the authorization middleware.